hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/NetworkTransactionGetReceiptHandler.java

```diff
@@ -55,16 +69,82 @@ public Response createEmptyResponse(@NonNull final ResponseHeader header) {
         return Response.newBuilder().transactionGetReceipt(response).build();
     }
 
+    @Override
+    public boolean requiresNodePayment(@NonNull ResponseType responseType) {
+        return responseType == ANSWER_ONLY || responseType == ANSWER_STATE_PROOF;
+    }
+
+    @Override
+    public boolean needsAnswerOnlyCost(@NonNull ResponseType responseType) {
+        return COST_ANSWER == responseType;
+    }
+
     @Override
     public void validate(@NonNull final QueryContext context) throws PreCheckException {
         requireNonNull(context);
-        throw new UnsupportedOperationException("Not implemented");
+        final var query = context.query();
+        final var recordCache = context.createStore(RecordCache.class);
+        final TransactionGetReceiptQuery transactionGetReceiptQuery = query.transactionGetReceiptOrThrow();
+        if (transactionGetReceiptQuery.hasTransactionID()) {
+            final var receipts =
+                    recordCache.getReceipt(transactionGetReceiptQuery.transactionIDOrElse(TransactionID.DEFAULT));
+            mustExist(receipts, INVALID_TRANSACTION_ID);
+        } else {
+            throw new PreCheckException(INVALID_TRANSACTION_ID);
+        }
     }
 
     @Override
     public Response findResponse(@NonNull final QueryContext context, @NonNull final ResponseHeader header) {
         requireNonNull(context);
         requireNonNull(header);
-        throw new UnsupportedOperationException("Not implemented");
+        final var query = context.query();
+        final var recordCache = context.createStore(RecordCache.class);
+        final var transactionGetReceiptQuery = query.transactionGetReceiptOrThrow();
+        final var responseBuilder = TransactionGetReceiptResponse.newBuilder();
+        final var transactionId = transactionGetReceiptQuery.transactionIDOrElse(TransactionID.DEFAULT);
```

Comment:
Is it even possible to get the "orElse" here? Or is it the case that by the time we get to this method we know for certain that the transaction ID exists because we did the check in validate? 

Things I'm worried about:
- If we use transactionIDOrThrow, it may be safe now, but if it is ever not safe and we fail to find it during testing, we could get an NPE in production
- If we use transactionIDOrElse(DEFAULT), then will the response be correct? If the transactionID isn't valid, we should return "INVALID_TRANSACTION_ID", but if the transactionID is valid but doesn't have a receipt, we return "RECEIPT_NOT_FOUND".

The second bug would be harder to find but less catastrophic. If we explicitly test this case, then going with the orThrow is probably preferable.

Reply:
ok done.